### ISSUE TITLE
`diskoLib.vmToolsSupportsCustomQemu` might test incorrect field

### ISSUE THREAD

**Issue by RadxaYuntian** [nix-community/disko#904]
Introduced in #851, `diskoLib.vmToolsSupportsCustomQemu` is used to check if the current `nixpkgs` supports overriding QEMU package.

While `disko` in general should be used with `nixos-unstable`, due to #900 I'm current targetting `nixos-24.11`. With the following [test project](https://github.com/RadxaYuntian/disko-test/commit/4b2f1cc848c239bd60b1f79b4d6fce667e6b32d1) I got the following error:

```
       error:
       Failed assertions:
       - You have set config.disko.imageBuild.qemu, but vmTools in your nixpkgs version "24.11.20241123.0c58267"
       does not support overriding the qemu package with the customQemu option yet.
       Please upgrade nixpkgs so that `lib.version` is at least "24.11.20240709".
```

Here is the table for the tested combinations:

|                   | nixos-24.11 | nixos-unstable |
|-------------------|-------------|----------------|
| diskoImages       | This issue  | OK             |
| diskoImagesScript | This issue  | #900           |

Checking `lib.versionAtLeast "24.11.20241123.0c58267" "24.11.20240709"` in `nix repl` got the expected `True`. After checking the source I found out that the assert is printing `lib.version` but `vmToolsSupportsCustomQemu` is checking `pkgs.lib.version`. I [updated the assert message](https://github.com/RadxaYuntian/disko/commit/7282c989c39ddbd7c6af08d25be7dc850e40ed5c) and the error now makes more sense:

```
       error:
       Failed assertions:
       - You have set config.disko.imageBuild.qemu, but vmTools in your nixpkgs version "24.11pre-git"
       does not support overriding the qemu package with the customQemu option yet.
       Please upgrade nixpkgs so that `lib.version` is at least "24.11.20240709".
```

I think we should change `vmToolsSupportsCustomQemu` to check `lib.version` as well, because that value looks to be more correct, but I know little between the difference of those 2 fields and how they actually should be used.

**Comment by RadxaYuntian** [comment]
Ping @iFreilicht who is the author of the original check.

**Comment by iFreilicht** [comment]
Can confirm. I looked into what could be causing the issue, and encountered something weird. I've downloaded your example flake, and it seems the `pkgs` that arrives in the configuration is different from the flake's input:

```
$ nix repl 
Lix 2.91.1
Type :? for help.
nix-repl> :lf .
Added 9 variables.

nix-repl> inputs.nixpkgs.lib.version
"24.11.20241123.0c58267"

nix-repl> inputs.nixpkgs.legacyPackages.x86_64-linux.lib.version
"24.11.20241123.0c58267"

nix-repl> inputs.disko.inputs.nixpkgs.lib.version
"24.11.20241123.0c58267"

nix-repl> nixosConfigurations.test.pkgs.lib.version
"24.11pre-git"

nix-repl> nixosConfigurations.test.config.disko.imageBuilder.pkgs.lib.version
"24.11pre-git
```

Either way, the solution is as you say, just use `lib` instead of `pkgs.lib`. Fix is in #906
